ami/trainers/i_jepa_latent_visualization_decoder_trainer.py

```diff
@@ -0,0 +1,167 @@
+# Ref: https://github.com/facebookresearch/ijepa
+
+from functools import partial
+from pathlib import Path
+from typing import Literal
+
+import torch
+import torchvision
+from torch.optim import Optimizer
+from torch.utils.data import DataLoader
+from typing_extensions import override
+
+from ami.data.buffers.buffer_names import BufferNames
+from ami.data.buffers.random_data_buffer import RandomDataBuffer
+from ami.data.interfaces import ThreadSafeDataUser
+from ami.models.i_jepa import IJEPAEncoder
+from ami.models.i_jepa_latent_visualization_decoder import (
+    IJEPALatentVisualizationDecoder,
+)
+from ami.models.model_names import ModelNames
+from ami.models.model_wrapper import ModelWrapper
+from ami.tensorboard_loggers import StepIntervalLogger
+
+from .base_trainer import BaseTrainer
+
+
+class IJEPALatentVisualizationDecoderTrainer(BaseTrainer):
+    def __init__(
+        self,
+        partial_dataloader: partial[DataLoader[torch.Tensor]],
+        partial_optimizer: partial[Optimizer],
+        device: torch.device,
+        logger: StepIntervalLogger,
+        encoder_type: Literal[ModelNames.I_JEPA_TARGET_ENCODER, ModelNames.I_JEPA_CONTEXT_ENCODER],
+        max_epochs: int = 1,
+        minimum_dataset_size: int = 1,
+        minimum_new_data_count: int = 0,
+        moving_average_rate: float = 0.999,
+    ) -> None:
+        """Initializes an IJEPALatentVisualizationDecoderTrainer object.
+
+        Args:
+            partial_dataloader: A partially instantiated dataloader lacking a provided dataset.
+            partial_optimizer: A partially instantiated optimizer lacking provided parameters.
+            device: The accelerator device (e.g., CPU, GPU) utilized for training the model.
+            encoder_type: Encoder to be visualized.
+            minimum_new_data_count: Minimum number of new data count required to run the training.
+        """
+        super().__init__()
+        self.partial_optimizer = partial_optimizer
+        self.partial_dataloader = partial_dataloader
+        self.device = device
+        self.logger = logger
+        self.encoder_type = encoder_type
+        self.max_epochs = max_epochs
+        self.minimum_dataset_size = minimum_dataset_size
+        self.minimum_new_data_count = minimum_new_data_count
+        self.moving_average_rate = moving_average_rate
+
+    def on_data_users_dict_attached(self) -> None:
+        self.image_data_user: ThreadSafeDataUser[RandomDataBuffer] = self.get_data_user(BufferNames.IMAGE)
+
+    def on_model_wrappers_dict_attached(self) -> None:
+        self.i_jepa_encoder: ModelWrapper[IJEPAEncoder] = self.get_training_model(self.encoder_type)
```

Comment:
ここは`get_frozen_model`を使います。
```suggestion
        self.i_jepa_encoder: ModelWrapper[IJEPAEncoder] = self.get_frozen_model(self.encoder_type)
```